### PR TITLE
Make AppHost/CLI incompatibility error specific and actionable

### LINKED ISSUE #10559: Improve Aspire compatibility error message to be more specific and actionable
As a follow-up to #9874, the compatibility error message should be more specific about which component needs to be updated (app host or Aspire CLI) rather than saying "upgrade the app host or aspire cli".

Current Behavior

The error message currently says something like:
> "upgrade the app host or aspire cli"

This is ambiguous and doesn't provide clear guidance on what action to take.

Proposed Improvement

1. Be specific: The error should clearly state which component needs updating based on the actual compatibility issue.

2. Provide actionable commands:
- If it's the app host version that needs updating, show the exact `dotnet add package` command to run
- Example: `dotnet add package Aspire.Hosting --version 9.4.0`

3. Consider automation: Even better, the error could prompt the user asking if they want to automatically update the package, then run the command for them.

Example of improved error messages

When App Host needs updating:

Error: Version mismatch detected.
Your Aspire.Hosting package version (9.3.0) is incompatible with Aspire CLI (9.4.0).

To fix this, run:
  dotnet add package Aspire.Hosting --version 9.4.0

Would you like to run this command now? [Y/n]

When CLI needs updating:

Error: Version mismatch detected.
Your Aspire CLI version (9.3.0) is incompatible with the project's Aspire.Hosting package (9.4.0).

To fix this, update the Aspire CLI:
  dotnet tool update aspire --version 9.4.0

Would you like to run this command now? [Y/n]

Benefits

- Reduces friction for developers
- Clear actionable steps to resolve the issue
- Less time spent figuring out what needs to be updated
- Better developer experience overall
- Consistent with other dotnet tooling that provides fix suggestions

Additional Considerations

- The error handler should detect which component is out of date and provide the appropriate message
- If the user chooses to auto-update, provide clear success/failure feedback
- Consider color coding for better visibility (red for errors, yellow for commands, green for success)

Related Issues

- #9874 - Improve the compatibility error message to include the Aspire.Hosting package version

### PR DESCRIPTION
**Pull request by shauryalowkeygotaura**
## Description

Fixes #10559

When the AppHost and CLI are incompatible, the CLI showed:

> ❌ The AppHost version is not compatible. Please upgrade the AppHost or Aspire CLI.

which doesn't say which side is out of date or what to do about it. This change compares the AppHost's Aspire version with the CLI version in `DisplayIncompatibleVersionError` and:

- **AppHost older than CLI** → "The AppHost is using an older Aspire version than the Aspire CLI. Update the AppHost's Aspire packages to continue." + `To update, run: aspire update`
- **CLI older than AppHost** → "The Aspire CLI is older than the Aspire version used by the AppHost. Update the Aspire CLI to continue." + the detected self-update command (dotnet tool / npm, reusing the same detection chain as `CliUpdateNotifier`, falling back to `aspire update`)
- **Version unparseable** (e.g. callers that pass the required capability when no version is available) → falls back to the existing generic message, no update command

The existing version/capability detail lines are unchanged. The update command line reuses the existing localized `ToUpdateRunCommand` string, so the only new localization entries are the two headline strings (xlf regenerated by the build).

## Testing

- Added 3 tests to `ConsoleInteractionServiceTests`: AppHost-older, CLI-older, and unparseable-version fallback
- `dotnet test --project tests/Aspire.Cli.Tests -- --filter-class "*ConsoleInteractionServiceTests"`: all passing locally

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by shauryalowkeygotaura** [comment]
@microsoft-github-policy-service agree

File: tests/Aspire.Cli.Tests/Interaction/ConsoleInteractionServiceTests.cs [hunk 1]
```diff
@@ -515,6 +515,84 @@ public void DisplayIncompatibleVersionError_WithMarkupCharactersInVersion_DoesNo
...
+        var console = AnsiConsole.Create(new AnsiConsoleSettings
+        {
+            Ansi = AnsiSupport.No,
+            ColorSystem = ColorSystemSupport.NoColors,
+            Out = new AnsiConsoleOutput(new StringWriter(output))
+        });
+        console.Profile.Width = 512;
+
+        var interactionService = CreateInteractionService(console);
+        var ex = new AppHostIncompatibleException("Incompatible", "baseline.v2");
```
[the remark this answers is not in the document]
Comment (shauryalowkeygotaura) [reply tests/Aspire.Cli.Tests/Interaction/ConsoleInteractionServiceTests.cs]: Done in 401ff3c — extracted `CreateInteractionServiceWithOutputCapture()` and all four DisplayIncompatibleVersionError tests now use it.

File: src/Aspire.Cli/Interaction/ConsoleInteractionService.cs [hunk 2]
```diff
@@ -442,12 +443,42 @@ public int DisplayIncompatibleVersionError(AppHostIncompatibleException ex, stri
...
+                updateCommand = "aspire update";
+            }
+            else if (comparison > 0)
+            {
+                errorMessage = InteractionServiceStrings.AppHostNotCompatibleUpdateCli;
+                updateCommand = DotNetToolDetection.GetDotNetToolUpdateCommand()
+                    ?? NpmInstallDetection.GetNpmUpdateCommand()
+                    ?? "aspire update";
+            }
+        }
```
[the remark this answers is not in the document]
Comment (shauryalowkeygotaura) [reply src/Aspire.Cli/Interaction/ConsoleInteractionService.cs:467]: Done in 401ff3c — added `DisplayIncompatibleVersionError_SameVersion_ShowsGenericMessageWithoutUpdateCommand` covering the equal-version case.

File: src/Aspire.Cli/Resources/xlf/InteractionServiceStrings.zh-Hans.xlf [hunk 3]
```diff
@@ -17,6 +17,16 @@
...
+      <trans-unit id="AppHostNotCompatibleUpdateAppHost">
+        <source>The AppHost is using an older Aspire version than the Aspire CLI. Update the AppHost's Aspire packages to continue.</source>
+        <target state="new">The AppHost is using an older Aspire version than the Aspire CLI. Update the AppHost's Aspire packages to continue.</target>
+        <note />
+      </trans-unit>
+      <trans-unit id="AppHostNotCompatibleUpdateCli">
+        <source>The Aspire CLI is older than the Aspire version used by the AppHost. Update the Aspire CLI to continue.</source>
+        <target state="new">The Aspire CLI is older than the Aspire version used by the AppHost. Update the Aspire CLI to continue.</target>
+        <note />
+      </trans-unit>
```
[the remark this answers is not in the document]
Comment (shauryalowkeygotaura) [reply src/Aspire.Cli/Resources/xlf/InteractionServiceStrings.zh-Hans.xlf:29]: This matches the repo's existing convention: these entries were generated by the build (XliffTasks), and `InteractionServiceStrings.zh-Hans.xlf` on main already contains 13 `state="new"` entries with the English source copied to the target. The localization pipeline picks up `state="new"` entries for translation, so I've left the generated output as-is.

**Comment by JamesNK** [comment]
# PR Testing Report

## PR Information
- **PR Number:** #18067
- **Title:** Make AppHost/CLI incompatibility error specific and actionable
- **Head Commit:** `401ff3c4989e1e1b982e23ea9db69805cbf54050`
- **Tested At:** 2026-06-10
- **Note:** CI artifacts were unavailable (workflow in `action_required` state), so testing was done by building from the PR branch locally.

## CLI Version Verification
- **Expected Commit:** `401ff3c4989e1e1b982e23ea9db69805cbf54050`
- **Local Build Version:** `13.5.0-dev`
- **Status:** :white_check_mark: Built from PR branch at expected commit

## Test Scenarios Executed

### Scenario 1: AppHost OLDER than CLI
**Status:** :white_check_mark: Passed

When the AppHost version (9.1.0) is older than the CLI (13.5.0-dev), the new message correctly shows:
```
:x: The AppHost is using an older Aspire version than the Aspire CLI. Update the AppHost's Aspire packages to continue.

        Aspire.Hosting Version: 9.1.0
        Aspire CLI Version: 13.5.0-dev
        Required Capability: baseline.v2

To update, run: aspire update
```

### Scenario 2: CLI OLDER than AppHost
**Status:** :white_check_mark: Passed

When the AppHost version (999.0.0) is newer than the CLI (13.5.0-dev), the new message correctly shows:
```
:x: The Aspire CLI is older than the Aspire version used by the AppHost. Update the Aspire CLI to continue.

        Aspire.Hosting Version: 999.0.0
        Aspire CLI Version: 13.5.0-dev
        Required Capability: baseline.v2

To update, run: aspire update
```

### Scenario 3: Unparseable Version (Generic Fallback)
**Status:** :white_check_mark: Passed

When the version string can't be parsed as semver (e.g., `baseline.v2`), the original generic message is shown without an update command:
```
:x: The AppHost version is not compatible. Please upgrade the AppHost or Aspire CLI.

        Aspire.Hosting Version: baseline.v2
        Aspire CLI Version: 13.5.0-dev
        Required Capability: baseline.v2
```

## Full Unit Test Results
- **Total:** 98 tests in `ConsoleInteractionServiceTests`
- **Passed:** 98
- **Failed:** 0
- **Duration:** ~1.5s

## Summary
| Scenario | Status | Notes |
|----------|--------|-------|
| AppHost older than CLI | :white_check_mark: Passed | Shows specific "update AppHost" message + `aspire update` command |
| CLI older than AppHost | :white_check_mark: Passed | Shows specific "update CLI" message + detected update command |
| Unparseable version | :white_check_mark: Passed | Falls back to generic message, no update command shown |
| Full test suite | :white_check_mark: Passed | 98/98 tests pass |

## Overall Result
**:white_check_mark: PR VERIFIED** - The incompatibility error messages are specific, actionable, and the fallback behavior is preserved for edge cases.

**Review by JamesNK — APPROVED** [review]
LGTM. The version comparison logic is correct, follows the existing detection chain pattern from `CliUpdateNotifier`, handles all edge cases (unparseable versions, same version), and has comprehensive test coverage for each branch.

**Comment by shauryalowkeygotaura** [comment]
The `VS Code extension E2E (Windows, zero-to-running)` failure looks unrelated to this change: the scenario itself passed (`✔ creates a new AppHost, adds a package, and debugs to the dashboard`) and only the "after each" teardown hook failed (`AggregateError: Zero-to-running E2E teardown failed.` in `fixtures.js`). This PR only touches the incompatible-version error message in `ConsoleInteractionService` plus two new localized strings, a path the E2E run never hits since the CLI and AppHost are built from the same commit. Could a maintainer re-run the failed job when convenient?
